sagemathinc/cocalc
# webapp/project files: library and +new

# Description
Reconcile +New contols in Files and fix some layout issues...

# Testing Steps
1. no files
    * create button → dialog
    * searching → create, also some buttons
1. there are files
    *  no search text, +create button
    * some other text in search → create file mode
1. editing the filename works, creating a file should show up with the indicated filetype
1. attempt to create a file without extension → error shows up. 
   *  similarly an error for  a stopped project. ← **that's no longer the case**. looks like this is also broken in production, though. this should be a follow up issue.
1. uploading 
   * by dropping onto existing files
   *  by dropping onto drop area in +create panel
   * in both cases the files listing updates 
1. actions that previously opened up +new should no longer do so (e.g. in jupyter menu, file → new)
1. library shows up and hides when clicking on close
1. grabbing a copy from the library works as usual, and the files listing switches to the target directory
1. settings panel boxes in project settings show up fine

# Relevant Issues

### [Checklist](https://github.com/sagemathinc/cocalc/wiki/PR-Checklist):

Front end:

### basic layout 

![screenshot from 2018-12-13 21-09-45](https://user-images.githubusercontent.com/207405/49964938-0b62ed80-ff1c-11e8-8af6-b8345aac25d9.png)

### create panel

![screenshot from 2018-12-13 21-07-55](https://user-images.githubusercontent.com/207405/49964941-0bfb8400-ff1c-11e8-8bc1-5ffe81bc4aef.png)

### no files, typing in search text

![screenshot from 2018-12-13 21-11-14](https://user-images.githubusercontent.com/207405/49964936-0b62ed80-ff1c-11e8-98ab-058d0f4732de.png)

### file action delete opened

![screenshot from 2018-12-13 21-09-10](https://user-images.githubusercontent.com/207405/49964939-0b62ed80-ff1c-11e8-96a9-000576294ae6.png)

### library

![screenshot from 2018-12-13 21-08-16](https://user-images.githubusercontent.com/207405/49964940-0bfb8400-ff1c-11e8-9148-aa2480d758e5.png)

### terminal output
* also still scrollable, aligns on the right hand side, changed the clear button

![screenshot from 2018-12-13 21-09-58](https://user-images.githubusercontent.com/207405/49964937-0b62ed80-ff1c-11e8-8a0f-627fb4ab8d34.png)

### placement of files button and dropdown changed slightly

![screenshot-cocalc com-2018 12 13-21-11-53](https://user-images.githubusercontent.com/207405/49964956-17e74600-ff1c-11e8-8d1b-1364bf5b4331.png)

### upload done + showing the uploaded file


![screenshot from 2018-12-13 21-18-53](https://user-images.githubusercontent.com/207405/49965191-c68b8680-ff1c-11e8-8597-7c55ee5f97ad.png)
